packages/app/cypress/e2e/debug.cy.ts

```diff
@@ -0,0 +1,151 @@
+import RelevantRunsDataSource_RunsByCommitShas from '../fixtures/gql-RelevantRunsDataSource_RunsByCommitShas.json'
+
+Cypress.on('window:before:load', (win) => {
+  win.__CYPRESS_GQL_NO_SOCKET__ = 'true'
+})
+
+// These mocks all the responses so we can get deterministic
+// results to test the debug page.
+// The JSON fixtures were generated by using a real app and capturing
+// the responses.
+describe('App - Debug Page', () => {
+  beforeEach(() => {
+    cy.scaffoldProject('cypress-in-cypress')
+    cy.openProject('cypress-in-cypress')
+    cy.startAppServer('component')
+
+    cy.loginUser()
+    cy.withCtx((ctx) => {
+      ctx.git?.__setGitHashesForTesting(['commit1', 'commit2'])
+    })
+
+    cy.remoteGraphQLIntercept((obj, _testState, options) => {
+      if (obj.operationName === 'RelevantRunsDataSource_RunsByCommitShas') {
```

Comment:
Why do we need to check for this query specifically when it looks like the other ones can be intercepted with `cy.intercept` since we're not using websockets during the test?

Reply:
I have the same question as Adam. Is this needed because it fires before the setting of the `__CYPRESS_GQL_NO_SOCKET__` ?  

I do like the ability to pass the fixtures through to the `removeGraphQLIntercept`

Reply:
I will bring this up at the "graphql e2e paint points" meeting, but basically `cy.intercept` is only for requests from the **front end** via `fetch`.

This means `cy.intercept` cannot work with subscriptions, those are always over websockets. What happens here is

1. subscription (via websocket)
2. remote graphql request (from server)

The only way to control the response for that is to use `remoteGraphQLIntercept` (this function is for intercepting request on the **server**). Despite the name, this is something we implemented specifically for GraphQL and it does **not** match the same signature as `cy.intercept` (but it could, and probably should).